pkg/apis/core/validation/validation.go

```diff
@@ -8121,3 +8138,56 @@ func validateNodeSelectorTermHasOnlyAdditions(newTerm, oldTerm core.NodeSelector
 	}
 	return true
 }
+
+var validSupplementalGroupsPolicies = sets.New(core.SupplementalGroupsPolicyMerge, core.SupplementalGroupsPolicyStrict)
+
+func validateSupplementalGroupsPolicy(supplementalGroupsPolicy *core.SupplementalGroupsPolicy, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+	if !validSupplementalGroupsPolicies.Has(*supplementalGroupsPolicy) {
+		allErrors = append(allErrors, field.NotSupported(fldPath, supplementalGroupsPolicy, sets.List(validSupplementalGroupsPolicies)))
+	}
+	return allErrors
+}
+
+func validateContainerStatusUsers(containerStatuses []core.ContainerStatus, fldPath *field.Path, podOS *core.PodOS) field.ErrorList {
+	allErrors := field.ErrorList{}
+	osName := core.Linux
+	if podOS != nil {
```

Comment:
I'm not super familiar with `core.PodOS` - when is it set?  What happens if it is not set?  Can we get here with this field being nil?  I don't see where scheduler considers it, and I don't see any default value defined.

@dchen1107

Reply:
It looks like this was introduced to detect Windows in admission plugins. https://github.com/kubernetes/enhancements/pull/2803/files
Scheduler interaction was marked as out of scope in that KEP.

Reply:
So this field is truly optional and advisory.  OK.